src/runtime/generators.js

```diff
@@ -147,7 +182,16 @@
         ctx.GState = ST_EXECUTING;
         ctx.action = action;
         ctx.sent = x;
-        var value = moveNext(ctx);
+        var value;
+        try {
```

Comment:
It would be worth thinking about if we could achieve this without try/catch? I feel like we should be able to know what the next state would be (we know that for throw).

Anyway. Not for this PR.